File: tools/video-builder/test-veo3.py
import os
import time

from google import genai
from google.genai import types


def main():
    api_key = os.getenv("GEMINI_API_KEY")
    if not api_key:
        raise SystemExit("GEMINI_API_KEY environment variable not set")

    client = genai.Client(api_key=api_key)

    print("[veo3-test] Submitting sample job...")
    operation = client.models.generate_videos(
        model="veo-3.0-generate-preview",
        prompt="a close-up shot of a golden retriever playing in a field of sunflowers",
        config=types.GenerateVideosConfig(
            negative_prompt="barking, woofing",
        ),
    )

    while not operation.done:
        print("[veo3-test] Waiting for completion...")
        time.sleep(20)
        operation = client.operations.get(operation)

    print("[veo3-test] Downloading result...")
    generated_video = operation.result.generated_videos[0]
    client.files.download(file=generated_video.video)
    generated_video.video.save("veo3_sample.mp4")
    print("[veo3-test] Saved veo3_sample.mp4")
if __name__ == "__main__":
    main()

File: tools/video-builder/test-veo3.js
#!/usr/bin/env node
import 'dotenv/config';
import path from 'path';
import { fileURLToPath } from 'url';
import chalk from 'chalk';
import { GoogleGenAI } from '@google/genai';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function main() {
  const apiKey = process.env.GEMINI_API_KEY;
  if (!apiKey) {
    console.error(chalk.red('GEMINI_API_KEY environment variable not set.'));
    process.exit(1);
  }

  const ai = new GoogleGenAI({ apiKey });

  console.log('[veo3-test] Submitting sample job...');
  let operation = await ai.models.generateVideos({
    model: 'veo-3.0-generate-preview',
    prompt: 'an attractive lady in a two piece swim wear dancing on the beach on this hot summer day',
    config: {
      numberOfVideos: 1,
      negativePrompt: 'barking, woofing',
    },
  });

  while (!operation.done) {
    console.log('[veo3-test] Waiting for completion...');
    await new Promise((resolve) => setTimeout(resolve, 20000));
    operation = await ai.operations.getVideosOperation({ operation });
  }

  if (operation.error) {
    throw new Error(`VEO3 job failed: ${JSON.stringify(operation.error)}`);
  }

  const videos = operation.response?.generatedVideos;
  if (!videos || videos.length === 0) {
    throw new Error('VEO3 response did not include generated videos');
  }

  const outPath = path.join(__dirname, 'veo3_sample.mp4');
  await ai.files.download({
    file: videos[0],
    downloadPath: outPath,
  });

  console.log('[veo3-test] Saved', outPath);
}

main().catch((error) => {
  console.error(chalk.red(`[veo3-test] Failed: ${error.message}`));
  process.exit(1);
});
